src/include/cursor.i

```diff
@@ -354,16 +354,16 @@ __cursor_func_init(WT_CURSOR_BTREE *cbt, bool reenter)
 }
 
 /*
- * __cursor_row_slot_return --
- *     Return a row-store leaf page slot's K/V pair.
+ * __cursor_row_slot_key_return --
+ *     Return a row-store leaf page slot's Key.
  */
 static inline int
-__cursor_row_slot_return(WT_CURSOR_BTREE *cbt, WT_ROW *rip, WT_UPDATE *upd)
+__cursor_row_slot_key_return(
```

Comment:
In these places, we were finding the visible update in-memory and lookaside *before* populating the key. Without a key, search in lookaside would fail. So by splitting the function, I have populated the key before the search and then we populate the value based on the update we find.